components/onfleet/sources/common/base.ts

```diff
@@ -0,0 +1,95 @@
+import { axios } from "@pipedream/platform";
+import onfleet from "../../app/onfleet.app";
+
+export default {
+  props: {
+    onfleet,
+    db: "$.service.db",
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+    webhookName: {
+      propDefinition: [
+        onfleet,
+        "webhookName",
+      ],
+    },
+  },
+  hooks: {
+    async deploy() {
+      await this.startDeploy();
+    },
+    async activate() {
+      // workaround - self call run() because createWebhook() can't be run on activate or deploy
+      await axios(this, {
+        url: this.http.endpoint,
+        method: "POST",
+        params: {
+          createHook: true,
+        },
+      });
+    },
+    async deactivate() {
+      const webhookId = this._getHookId();
+      if (webhookId) await this.onfleet.deleteWebhook(webhookId);
+    },
+  },
+  methods: {
+    _setHookId(hookId) {
+      this.db.set("hookId", hookId);
+    },
+    _getHookId() {
+      return this.db.get("hookId");
+    },
+    emitEvent(event) {
+      const meta = this.generateMeta(event.body);
+      this.$emit(event, meta);
+    },
+    async startDeploy() {
+      // Retrieve historical events
+      const items = this.onfleet.paginate({
+        fn: this.getFn(),
+        field: this.getField(),
+        maxResults: 25,
+      });
+      const resources = [];
+
+      for await (const item of items) {
+        resources.push(item);
+      }
+
+      resources
+        .reverse()
+        .forEach((resource) => {
+          this.$emit(resource, this.generateMeta(resource));
+        });
+    },
+    async createHook() {
+      const response = await this.onfleet.createWebhook({
+        url: this.http.endpoint,
+        name: this.webhookName,
+        trigger: this.getTrigger(),
+      });
+      this._setHookId(response.id);
+      return response;
+    },
+  },
+  async run(event) {
+    const responseData = {
+      status: 200,
+      body: String,
```

Comment:
Is this really meant to be the `String` constructor and not something like an empty string `""`?